### DISCUSSION TITLE
Script Kit 1.53.22 - April 2023 Release

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#1178] [Announcements]
# Script Kit 1.53.22 - April 2023 Release

- Download from https://www.scriptkit.com/
- Join our Discord: https://discord.gg/qnUX4XqJQd
- Sponsor to Keep Script Kit Growing ❤️ (and unlock the debugger): [https://github.com/sponsors/johnlindquist/sponsorships?sponsor=johnlindquist](https://github.com/sponsors/johnlindquist/sponsorships?sponsor=johnlindquist&tier_id=235205)

### `await mic()`

Using await mic will return a buffer of the audio recorded from your microphone.

```js
// Name: Transcribe Mic

import "@johnlindquist/kit"

let { Configuration, OpenAIApi } = await import("openai")

let config = new Configuration({
  apiKey: await env("OPENAI_API_KEY"),
})

let openai = new OpenAIApi(config)

let data = await mic()

let stream = Readable.from(data)
// https://github.com/openai/openai-node/issues/77#issuecomment-1463150451
stream.path = "speech.webm"

// If you're confused by "createTranscription" params, see: https://github.com/openai/openai-node/issues/93#issuecomment-1471285341
let response = await openai.createTranscription(stream, "whisper-1")

let transcriptionPath = tmpPath(`${Date.now()}.txt`)
await writeFile(transcriptionPath, response.data.text)

await editor(response.data.text)
```

![http://res.cloudinary.com/johnlindquist/image/upload/v1679783859/clipboard/pvbsvjjj1zrmlode3wwn.png](http://res.cloudinary.com/johnlindquist/image/upload/v1679783859/clipboard/pvbsvjjj1zrmlode3wwn.png)

### `await webcam()`

Using await webcam will return a buffer of the image captured from your webcam.

![http://res.cloudinary.com/johnlindquist/image/upload/v1679778480/clipboard/lxhy4fmndgzp7xximgco.png](http://res.cloudinary.com/johnlindquist/image/upload/v1679778480/clipboard/lxhy4fmndgzp7xximgco.png)

```js
// Name: Selfie

import "@johnlindquist/kit"

let buffer = await webcam()

let filePath = tmpPath("webcam.png")

await writeFile(filePath, buffer)
await revealFile(filePath)
```

### Resizing

The prompt will now grow and shink to match the length of the list of options.

You can also manually control the size of prompts by using the new `PROMPT` constants:

```js
await editor({
  width: PROMPT.WIDTH["5XL"],
  height: PROMPT.HEIGHT["5XL"],
})
```

### Terminal "Close on Exit"

The terminal will now automatically close when you exit the command.

> Note: By default, it start a shell (zsh/bash/cmd.exe for mac/linux/windows), so you'll need to run `&& exit` after your command

```js
await term("brew install ffmpeg && exit")
```

You can also use the `shell` option to disable the shell and just run the tool directly:

```js
await term({
  shell: false,
  command: "brew",
  args: ["services", "restart", "yabai"],
  closeOnExit: false, // optional, if you want to view output before it closes
})
```

> Note: Also fixed some resizing bugs with the terminal.

### Windows App Launcher

Press `;` from the main menu to launch the App Launcher.

Thanks to @dodgez: https://github.com/johnlindquist/kit/pull/1161

> Note: We still need a Linux App Launcher if anyone wants to take a stab at it :)

### Custom Fonts

You can now use custom fonts with the UI. In your `~/.kenv/.env` file, add:

```bash
KIT_MONO_FONT="Menlo"
KIT_SANS_FONT="Arial"
```

### Experimental: Shebang Scripts 🎉

Add a script in your `~/.kenv/scripts` directory with a shebang:

![http://res.cloudinary.com/johnlindquist/image/upload/v1679779537/clipboard/v1oja8v05dgexdgtfjba.png](http://res.cloudinary.com/johnlindquist/image/upload/v1679779537/clipboard/v1oja8v05dgexdgtfjba.png)

```bash
#!/bin/bash
open "https://scriptkit.com"
```

Then run it from the main menu

### Experimental: Windows `.bat` Files

Mac and Linux users have always had executables they could run in the terminal in the `~/.kenv/bin` dircetory. Now, on Windows, Script Kit will generate `.bat` files associated with each script that you can run from the terminal.

### Info Choices

We now have a custom "info" choice that isn't selectable, but shows up in the list to present the user with additional information about the input/list:

![http://res.cloudinary.com/johnlindquist/image/upload/v1679783784/clipboard/j4av6h2ybe1vezln2tt3.png](http://res.cloudinary.com/johnlindquist/image/upload/v1679783784/clipboard/j4av6h2ybe1vezln2tt3.png)

```js
// Name: Testing Info OnNoChoices

import "@johnlindquist/kit"

await arg("Select a number",
  [
    {
      name: "Sorry, no matches",
      info: "onNoChoices", // or "always"
    },
    "one",
    "two",
    "three",
  ]
)
```

### Removing `npm` requirement

When you run a script with a missing module, it will catch the error and prompt you to install it. This removes the requirement of the `await npm()`, but you can still use it if you find it more convenient.

The examples and scripts that I share from now on will no longer use `await npm()`, but I have no plans to deprecate it.

## Fixes

- Lots of little improvements around the `chat` component.
- Some solid performance improvements
- Upgrading to lots of the latest libraries

## Call for Help: Disable Window Animation on Windows?

If anyone knows how to disable the window animation on Windows (the one that transparently zooms in when you open a window), please let me know because I think it's super annoying and I'd love to disable it in the app. For now, I strongly recommend disabling it globally on Windows:

1. Open the Settings app by pressing the Windows key + I.
2. Click on “Ease of Access”.
3. Scroll down to “Other options” and click on “Visual options”.
4. Under “Play animations in Windows”, toggle the switch to “Off”.

## One Last Thing...

Script Kit AI course part 1 dropping this week... 💥

**Comment by johnlindquist** [comment]
### Launch bugs 🐞 (will have fixed this weekend):

1. Found that the new "clipboard history" lib fails to "spawn" on mac x86 machines:
https://github.com/johnlindquist/kit/discussions/1179

2. Some "share" script screens don't respect the new resize logic (you'll just have to manually drag a corner)

Please report any more if you find them and I'll get it sorted by this weekend 🙏

**Comment by nbbaier** [comment]
I've been looking for the Script Kit AI course but haven't been able to find it. Is it available yet?

**Reply by johnlindquist** [reply]
@nbbaier Not yet... I've got ~40 lessons done and we're adding some course features to scriptkit.com to be able to host it. It's  coming together really nicely and getting closer every day :)

**Reply by nbbaier** [reply]
Looking forward to seeing it when it comes together! I really love Script Kit.
